### PR TITLE
NXP-23440: initial checkin

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

**Review by kevinleturc — COMMENTED** [review]
👍

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.nuxeo</groupId>
+    <artifactId>nuxeo-addons-parent</artifactId>
+    <version>9.10-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.nuxeo.jsf.to.web.ui</groupId>
```
Comment (efge) [inline pom.xml]: This group seems a bit fancy. Please use a simple and existing one like `org.nuxeo.web.ui` instead (also update the other commit in addons)

**Review by efge — CHANGES_REQUESTED** [review]

File: src/main/java/org/nuxeo/jsf/to/web/ui/NuxeoJSFWebUIRedirectFilter.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thomas Roger
+ */
+
+package org.nuxeo.jsf.to.web.ui;
```
Comment (efge) [inline src/main/java/org/nuxeo/jsf/to/web/ui/NuxeoJSFWebUIRedirectFilter.java]: The Java package isn't very canonical either. This is supposed to be a hierarchical representation of where this lives in a hierarchy of packages.

File: src/main/java/org/nuxeo/jsf/to/web/ui/WebUIRedirectCodec.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        List<String> fragments = new ArrayList<>();
+        fragments.add(getPrefix());
+        if (idRef != null) {
+            fragments.add("doc");
+            fragments.add(docLoc.getServerName());
+            fragments.add(idRef.toString());
+        } else {
+            fragments.add("browse" + pathRef.toString());
+        }
+        return StringUtils.join(fragments, "/");
```
Comment (efge) [inline src/main/java/org/nuxeo/jsf/to/web/ui/WebUIRedirectCodec.java]: `String.join("/", fragments)`

**Review by ataillefer — APPROVED** [review]

**Comment by ataillefer** [comment]
:+1:

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      </snapshots>
+    </repository>
+    <repository>
+      <id>public-snapshot</id>
+      <url>http://maven.nuxeo.org/nexus/content/groups/public-snapshot</url>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+    </repository>
+  </repositories>
```
Comment (atchertchian) [inline pom.xml:43]: these are not inherited from parent confs?

**Review by atchertchian — APPROVED** [review]
:+1:, and i'm wondering if this could be useful on nxfile URLs too

**Comment by atchertchian** [comment]
:+1:

**Review by efge — APPROVED** [review]

**Comment by efge** [comment]
👍

File: pom.xml (hunk 4 above)
Comment (troger) [reply pom.xml:43]: You need them if you just checkout the `nuxeo-jsf-to-web-ui` module and want to build it, as you don't have any parent pom available...

**Comment by troger** [comment]
@atchertchian `nxfile` URLs are handled by the `DownloadServlet` in the `nuxeo-platform-web-common` module so no need to redirect them on web ui or something.

**Pull request merged**